src/services/rpcService.ts

```diff
@@ -1,134 +1,211 @@
-      try {
-        if (!validatePublicKey(publicKey)) {
-          return '';
-        }
-  
-        const lamports = amount * 1e9;
-        const signature = await this.connection.requestAirdrop(publicKey, lamports);
-        const latestBlockhash = await this.connection.getLatestBlockhash();
-        await this.connection.confirmTransaction({
-          signature,
-          ...latestBlockhash
-        });
-        console.log(`Debug: Successfully airdropped ${amount} SOL to ${publicKey.toBase58()}`);
-        return signature;
-      } catch (error) {
-        console.error("Error airdropping SOL:", error);
-        throw error;
+  }
+
+  /**
+   * Get the SOL balance for a given public key
+   * @param {PublicKey} publicKey - The public key to check the balance for
+   * @returns {Promise<bigint>} The SOL balance as a bigint
+   */
+  async getSolBalance(publicKey: PublicKey): Promise<bigint> {
+    try {
+      if (!validatePublicKey(publicKey)) {
+        return BigInt(0);
       }
+
+      console.log("Debug: getting SOL balance for", publicKey.toBase58());
+      const balance = await this.connection.getBalance(publicKey);
+      if (balance === 0) {
+        console.warn(`Debug: SOL balance for ${publicKey.toBase58()} is 0`);
+        return BigInt(0);
+      }
+      console.log(`Debug: SOL balance for ${publicKey.toBase58()} is ${balance / 1e9} SOL`);
+      return BigInt(balance);
+    } catch (error) {
+      console.error("Error fetching SOL balance:", error);
+      throw error;
     }
-  
-    /**
-     * Get the SOL balance for a given public key
-     * @param {PublicKey} publicKey - The public key to check the balance for
-     * @returns {Promise<BN>} The SOL balance as a BN
-     */
-    async getSolBalance(publicKey: PublicKey): Promise<BN> {
-      try {
-        if (!validatePublicKey(publicKey)) {
-          return new BN(0);
-        }
-  
-        console.log("Debug: getting SOL balance for", publicKey.toBase58());
-        const balance = await this.connection.getBalance(publicKey);
-        if (balance === 0) {
-          console.warn(`Debug: SOL balance for ${publicKey.toBase58()} is 0`);
-          return new BN(0);
+  }
+
+  /**
+   * Get the SPL token balance for a given public key and mint
+   * @param {PublicKey} publicKey - The public key to check the balance for
+   * @param {PublicKey} mint - The SPL token mint
+   * @param {PublicKey} programId - The SPL token program ID (default: TOKEN_PROGRAM_ID)
+   * @returns {Promise<bigint>} The SPL token balance as a bigint
+   */
+  async getSplBalance(publicKey: PublicKey, mint: PublicKey, programId = TOKEN_PROGRAM_ID): Promise<bigint> {
+    try {
+      const pk = new PublicKey(publicKey);
+      if (!validatePublicKey(pk)) {
+        return BigInt(0);
+      }
+
+      console.log("getting spl balance", mint);
+      const response = await this.connection.getTokenAccountsByOwner(
```

Comment:
We're following the Associated Token Account convention, so there's only ever one token account per mint per user.

I recommend using [this approach](https://github.com/solana-labs/solana-program-library/blob/d72289c79a04411c69a8bf1054f7156b6196f9b3/token/js/src/actions/getOrCreateAssociatedTokenAccount.ts#L51) to fetch the token account.